dianna/dashboard/pages/Time_series.py

```diff
@@ -51,25 +60,75 @@
         contributing positively towards the classification decision are
         indicated in red and those who contribute negatively in blue.
         """)
+    elif load_example == "Scientific case: FRB":
+        ts_file = (data_directory / 'FRB211024.npy')
+        ts_model_file = (model_directory /
+                        'apertif_frb_dynamic_spectrum_model.onnx')
+        ts_label_file = (label_directory / 'apertif_frb_classes.txt')
+
+        # FRB data must be preprocessed
+        def preprocess(data):
+            """Preprocessing function for FRB use case to get the data in the right shape."""
+            return np.transpose(data, (0, 2, 1))[..., None].astype(np.float32)
+
+        ts_data = open_timeseries(ts_file)
+        ts_data_dianna = ts_data.T[None, ...]
+        ts_data_model = ts_data[None, ..., None]
+
+        st.markdown(
+            """This example demonstrates the use of DIANNA
+            on a pre-trained binary classification model trained to classify
+            Fast Radio Burst (FRB) timeseries data.
+            The goal of the pre-trained convolutional neural network is to
+            determine whether or not the input data contains an
+            FRB-like signal, whereby the two classes are noise and FRB.
+            """)
+    else:
+        st.info('Select an example in the left panel to coninue')
+        st.stop()
+
+
+# Option to upload your own data
+if input_type == 'Use your own data':
+    load_example = None
+
+    ts_file = st.sidebar.file_uploader('Select input data',
+                                    type='npy')
+
+    ts_model_file = st.sidebar.file_uploader('Select model',
+                                            type='onnx')
+
+    ts_label_file = st.sidebar.file_uploader('Select labels',
+                                            type='txt')
+
+if input_type is None:
+    st.info('Select which input type to use in the left panel to continue')
+    st.stop()
+
 
 if not (ts_file and ts_model_file and ts_label_file):
     st.info('Add your input data in the left panel to continue')
     st.stop()
 
-ts_data = open_timeseries(ts_file)
+if load_example != "Scientific case: FRB":
+    ts_data_model = open_timeseries(ts_file)
+    ts_data_dianna = ts_data_model
```

Comment:
the same comment as above, the names `ts_data_model` and `ts_data_dianna` are not informative, could you rename these variables to something more descriptive?  Also, please add an inline comment explaining the purpose of defining these two variables.

Reply:
I renamed them to ts_data_predictor and ts_data_explainer respectively and added a comment to make it clearer